src/main/java/com/odysseusinc/arachne/portal/util/SubmissionHelper.java

```diff
@@ -421,6 +432,94 @@ public void updateExtendInfo(Submission submission) {
         }
     }
 
+    private class PathwaySubmissionExtendInfoStrategy extends SubmissionExtendInfoAnalyzeStrategy {
+
+        @Override
+        public void updateExtendInfo(Submission submission) {
+
+            JsonObject resultInfo = new JsonObject();
+            try {
+                final String rootDir = contentStorageHelper.getResultFilesDir(submission);
+                final String resultsDir = filePath(rootDir, "results");
+                final String designDir = filePath(rootDir, "design");
+
+                JsonElement design;
+                try(JsonReader jsonReader = new JsonReader(new InputStreamReader(contentStorageService.getContentByFilepath(designDir + PATH_SEPARATOR + "StudySpecification.json")))) {
+                    JsonParser parser = new JsonParser();
+                    design = parser.parse(jsonReader);
+                }
+                resultInfo.add("design", design);
+
+                // Pathway codes
+                String path = filePath(resultsDir, "pathway_codes.csv");
+                JsonArray pathwayCodes = parseCsv(path, rec -> {
+                    JsonObject pathwayCode = new JsonObject();
+                    pathwayCode.add("code", getJsonPrimitive(rec.get("CODE")));
+                    pathwayCode.add("name", getJsonPrimitive(rec.get("NAME")));
+                    pathwayCode.add("isCombo", getJsonPrimitive(rec.get("IS_COMBO")));
+                    return pathwayCode;
+                });
+                resultInfo.add("eventCodes", pathwayCodes);
+
+                path = filePath(resultsDir, "cohort_stats.csv");
+                JsonArray cohortStats = parseCsv(path, rec -> {
+                   JsonObject stat = new JsonObject();
+                   stat.add("targetCohortId", getJsonPrimitive(rec.get("TARGET_COHORT_ID")));
+                   stat.add("targetCohortCount", getJsonPrimitive(rec.get("TARGET_COHORT_COUNT")));
+                   stat.add("totalPathwaysCount", getJsonPrimitive(rec.get("PATHWAYS_COUNT")));
+                   return stat;
+                }, ArachneCollectors.toJsonArray());
+
+                // Cohort Paths
+                path = filePath(resultsDir, "pathway_results.csv");
+                Map<Integer, JsonArray> pathwayResults  = parseCsv(path, rec -> {
+                   JsonObject result = new JsonObject();
+                   result.add("targetCohortId", getJsonPrimitive(rec.get("TARGET_COHORT_ID")));
+                   result.add("personCount", getJsonPrimitive(rec.get("COUNT_VALUE")));
+                   List<String> cohortPath = new ArrayList<>();
+                   for(int i = 1; i <= 10; i++) {
```

Comment:
magic number, if it is really necessary to keep it, then create a constant with descriptive name